Ft login page bug fix

GilbertUp opened the pull request:
### What does this PR do?

It allows registered user to login

### How should this be tested?
-Clone this [repo](https://github.com/atlp-rwanda/Dominators-Phantom-FE/compare/develop...ft-login-page-bug-fix)
-checkout to the branch called ft-login-page-bud-fix
-Run the command npm i and later npm start
-click on login and in put admin7@test.com and pass12345 as user admin  credential
### What to expect?
-if the email is in invalid format or missing, an appropriate message will be displayed
-if the password is missing or less than 8 characters, an appropriate message will be displayed
-For successful attempt you will be redirected to admin dashboard and error messages are displayed for unsuccessful attempts
### Acceptance criteria

- [x] Appropriate error messages are displayed on invalid inputs (Form validation).
- [x] Appropriate error messages are displayed on unsuccessful attempts
- [x]  Upon successful login, the user should be redirected to their notification page (At this point in time, the notification page might be a black page).
- [ ]  Skeleton UIs/screen are/is shown when the page loads/is loading.
- [x]  The page is responsive
- [ ]  Each user  role upon successful login should be redirected to their respective profile page

furebo commented on src/components/Login/LoginComponent.js:
```diff
@@ -0,0 +1,161 @@
+  })
+
+  const handleLogin = (values)=>{
+    console.log(db)
```
Please remove the console.log line.

furebo commented on src/components/Login/LoginComponent.js:
```diff
@@ -0,0 +1,161 @@
+      return res.json();
+    })
+    .then ((result)=>{
+      console.log(result)
```
Please remove also the console.log line

furebo reviewed (changes requested):
Remove the console.log lines and squash the commits to one commit

Anguandia commented:
Please note the following:

@GilbertUp good PR description. Much as you are now in the backend, please address the following
- squash these commits, 38 is unacceptable
- was this a bug or a feature? the branch name is confusing. Please rename it following the team conventions and best practices documented [here](https://github.com/atlp-rwanda/engineering-playbook/tree/master/5.%20Developing/Conventions#branch-naming)
- despite the spot-on PR description, the formatting is still wanting, please acquaint yourself with markdown syntax. you can search online

@gerard1001  you now own this task, please respond to any technical review comments/change requests to this PR. Also address the following:
- write unit tests for the feature
- test setup is wanting, you may start from setting up testing

Anguandia commented:
A few more observations:
- please change the cursor to pointer when hovering over the login button
- it will be good to read the backend base url from the environment so that different instances may be used in different environments

gerard1001 committed:
deleted a dummy user

rebasing login

rebase continued

re-added login component

updated media queries for right side bar

updated media queries for right side bar

moved backend url to the environment

gerard1001 committed:
deleted a dummy user

furebo commented on src/components/Login/LoginComponent.js:
```diff
@@ -0,0 +1,174 @@
+            result.data.user.role == "driver"
+          ) {
+            redirect("/Dashboard");
+          } else {
+          }
```
Please remove the empty else block

furebo reviewed (changes requested):
Hi good work Gelard but resolve the appearing conflicts do the requested change

Anguandia commented on src/utils/db.js:
```diff
@@ -1 +1 @@
-export const db = "http://localhost:4000";
+export const db = "http://localhost:4000/api/v1";
```
We need to dynamically inject this from the environment

Anguandia commented on package.json:
```diff
@@ -33,8 +34,8 @@
+    "sweetalert": "^2.1.2"
   },
   "devDependencies": {
     "@babel/core": "^7.17.9",
```
We also need to streamline project vs dev dependencies

Anguandia reviewed (approved):
Good work! just the 2 changes, we'll be good to go

gerard1001 committed:
updated right bar styles

gerard1001 committed:
updated right bar

gerard1001 committed:
re added login.css
